includes/class-wc-google-gtag-js.php

```diff
@@ -164,14 +164,14 @@ protected function add_transaction_enhanced( $order ) {
 		}
 		$items .= "]";
 
-		$code  = "" . self::tracker_var() . "( 'event', 'purchase', {
+		$code = self::tracker_var() . "( 'event', 'purchase', {
 			'transaction_id': '" . esc_js( $order->get_order_number() ) . "',
 			'affiliation': '" . esc_js( get_bloginfo( 'name' ) ) . "',
 			'value': '" . esc_js( $order->get_total() ) . "',
 			'tax': '" . esc_js( $order->get_total_tax() ) . "',
 			'shipping': '" . esc_js( $order->get_total_shipping() ) . "',
-			'currency': '" . esc_js( version_compare( WC_VERSION, '3.0', '<' ) ? $order->get_order_currency() : $order->get_currency() ) . "',  // Currency,
-			'items': " . $items . ",
+			'currency': '" . esc_js( $order->get_currency() ) . "',
+			'items': {$items},
```

Comment:
❓ Just curious, why we don't escape $items output with `esc_js` as well?

Reply:
We should only escape the data within the single quotes, not escape the full `$items` variable. If we look at the `$items` variable it comes from the function `add_item` which uses `esc_js` for each field individually, so no need to run it through `esc_js` a second time.